apps/app/src/components/QuestionItem/QuestionVoting.tsx

```diff
@@ -1,14 +1,72 @@
+"use client";
+
+import { useMutation, useQuery } from "@tanstack/react-query";
 import { twMerge } from "tailwind-merge";
+import { useSearchParams } from "next/navigation";
+import { useDevFAQRouter } from "../../hooks/useDevFAQRouter";
+import { useUser } from "../../hooks/useUser";
+import {
+	downvoteQuestion,
+	getQuestionsVotes,
+	upvoteQuestion,
+} from "../../services/questions.service";
 import { pluralize } from "../../utils/intl";
+import { Technology } from "../../lib/technologies";
+import { PAGE_SIZE } from "../../lib/constants";
+import { DEFAULT_SORT_BY_QUERY, getQuerySortBy } from "../../lib/order";
 
 type QuestionVotingProps = Readonly<{
-	votes: number;
-	voted: boolean;
+	questionId: number;
+	page: number;
+	technology: Technology;
 }>;
 
 const votesPluralize = pluralize("głos", "głosy", "głosów");
 
-export const QuestionVoting = ({ votes, voted }: QuestionVotingProps) => {
+export const QuestionVoting = ({ questionId, page, technology }: QuestionVotingProps) => {
+	const searchParams = useSearchParams();
+	const querySortBy = getQuerySortBy(searchParams.get("sortBy") || DEFAULT_SORT_BY_QUERY);
+
+	const { data, refetch } = useQuery({
+		queryKey: ["votes"],
```

Comment:
`queryKey` musi być unikalne dla konkretnego zapytania. To znaczy, że musi zawierać w sobie **wszystkie** zmienne, które przekazujesz do danego query: category, limit, offset, orderBy, order.